From af96fabbc418e2ca5408d40f9b7291bf937559a2 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Wed, 25 Jan 2017 10:49:37 +0100
Subject: [PATCH 9/9] core: avcodec requires 32-bytes memory alignment in many
 cases

For example a IA0L texture on Windows is not 32 bytes aligned and avcodec
complains about it. This will avoid disabling direct rendering in those cases
and reducing memory copies which is even more crucial for high bit depth that
usually come with high definition (4K HEVC 10 bits in this case).

--
replaces https://patches.videolan.org/patch/7971/
---
 src/misc/picture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/misc/picture.c b/src/misc/picture.c
index f6671edd51..5959a36abf 100644
--- a/src/misc/picture.c
+++ b/src/misc/picture.c
@@ -70,7 +70,7 @@ static int AllocatePicture( picture_t *p_pic )
         return VLC_ENOMEM;
     }
 
-    uint8_t *p_data = aligned_alloc( 16, i_bytes );
+    uint8_t *p_data = aligned_alloc( 32, i_bytes );
     if( i_bytes > 0 && p_data == NULL )
     {
         p_pic->i_planes = 0;
-- 
2.11.0

